source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.cc

```diff
@@ -43,17 +43,26 @@ Http::FilterHeadersStatus Http1BridgeFilter::encodeHeaders(Http::ResponseHeaderM
   if (doStatTracking()) {
     chargeStat(headers);
   }
+  if (!do_bridging_) {
+    return Http::FilterHeadersStatus::Continue;
+  }
 
-  if (!do_bridging_ || end_stream) {
+  response_headers_ = &headers;
+  if (end_stream) {
+    setupHttp1Status(headers.GrpcStatus(), headers.GrpcMessage());
```

Comment:
Can we add a runtime flag here since we're changing the behavior? https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md#runtime-guarding